// tooy-ui/dist/tooy-ui.js
import { computed as l, openBlock as d, createElementBlock as c, normalizeClass as u, unref as r, createElementVNode as _, renderSlot as i } from "vue";
const p = (t, e) => {
  const n = t.__vccOpts || t;
  for (const [s, a] of e)
    n[s] = a;
  return n;
}, b = ["disabled"], f = {
  name: "yButton"
}, y = /* @__PURE__ */ Object.assign(f, {
  props: {
    type: {
      type: String,
      default: "default"
    },
    disabled: Boolean,
    plain: Boolean
  },
  setup(t) {
    const e = t, n = l(() => [
      "y-button",
      `y-button-${e.type}`,
      e.disabled ? "y-button-disabled" : "",
      e.plain ? "y-button-plain" : ""
    ]);
    return (s, a) => (d(), c("button", {
      class: u(r(n)),
      disabled: t.disabled
    }, [
      _("span", null, [
        i(s.$slots, "default", {}, void 0, !0)
      ])
    ], 10, b));
  }
}), o = /* @__PURE__ */ p(y, [["__scopeId", "data-v-81eb8df8"]]);
o.install = (t) => {
  t.component(o.name, o);
};
const m = (t) => {
  t.use(o);
}, v = {
  install: m
};
export {
  v as default,
  o as yButton
};
